drivers/i2c/i2c_ll_stm32.meta.jinja2

```diff
@@ -0,0 +1,36 @@
+{# Copyright (c) 2018 Nordic Semiconductor ASA #}
+{# SPDX-License-Identifier: Apache-2.0         #}
+
+{% extends 'device_declare.jinja2' %}
+
+{% block init_params %}
+        {% do params.update({'irq_flag':'CONFIG_I2C_STM32_INTERRUPT'}) %}
+        {% do params.update({'config_struct':'i2c_stm32_config'}) %}
+        {% do params.update({'data_struct':'i2c_stm32_data'}) %}
+        {% do params.update({'api_struct':'api_funcs'}) %}
+        {% do params.update({'init_function':'i2c_stm32_init'}) %}
+{% endblock %}
+
+{# Because these blocks are "scoped" all variables in the macro from the
+   base template can be accessed in these blocks #}
+{% block config_struct_body %}
+        .i2c = (I2C_TypeDef *){{ "%x"|format(device['reg']['0']['address']) }},
+        .pclken = {
+                .enr = {{ device['clocks']['0']['bits'] }},
+                .bus = {{ device['clocks']['0']['bus'] }},
+        },
+#ifdef {{ get_param(params, 'irq_flag') }}
```

Comment:
For reader, it would be clearer to keep using CONFIG_I2C_STM32_INTERRUPT here.
Since we're templating the config_struct, only missing args should be the ones from EDTS.

Reply:
I don't like to have "multiple truths".

If this file was copied and used as a template for different metadata file, or extended/changed, the developer only need to set the irq_flag one place.

Reply:
Agreed, but this makes difficult to read and it is not strictly required.

In the end, solution might be to remove this element the structure and add an element in DEVICE_AND_API_INIT macro.

Reply:
Ok, just because you asked for it, I keep my eyes shut when I push enter after writing 'git commit':D